src/agent/coverage/src/cache.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeSet, HashMap};
+
+use crate::code::{ModuleIndex, ModulePath};
+use crate::disasm::ModuleDisassembler;
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct ModuleCache {
+    pub cached: HashMap<ModulePath, ModuleInfo>,
+}
+
+impl ModuleCache {
+    pub fn new() -> Self {
+        let cached = HashMap::new();
+
+        Self { cached }
+    }
+
+    pub fn fetch(&mut self, path: &ModulePath) -> Result<Option<&ModuleInfo>> {
+        if !self.cached.contains_key(path) {
+            self.insert(path)?;
+        }
+
+        Ok(self.cached.get(path))
+    }
+
+    pub fn insert(&mut self, path: &ModulePath) -> Result<()> {
+        let entry = ModuleInfo::new_elf(path)?;
```

Comment:
Should this function be Linux only, or will you have an equivalent Windows api so just this line needs to be conditionally compiled?

Reply:
We'll conditionally compile in some form (follow-up PR).